source/message/OP_MSG.rst

```diff
@@ -60,7 +60,8 @@ node supports ``OP_MSG``. Drivers that have only ever supported MongoDB 3.6 and
 newer MAY default to using ``OP_MSG``.
 
 If the node supports ``OP_MSG``, any and all messages MUST use ``OP_MSG``,
-optionally compressed with ``OP_COMPRESSED``.
+optionally compressed with ``OP_COMPRESSED``. This includes authentication
+messages.
```

Comment:
This is ambiguous. One possible reading is that authentication messages must use `OP_MSG` (which is true) and optionally be compressed with `OP_COMPRESSED` (which is false). Authentication-related messages MUST NEVER be compressed. They were explicitly excluded from compression because of concerns about unintentionally introducing security issues. The following is a list of commands that must never be compressed used by the .NET/C# driver:

hello
isMaster
saslStart
saslContinue
getnonce
authenticate
createUser
updateUser
copydbsaslstart
copydbgetnonce
copydb

If we want to reconsider compressing these messages we can do that, but it seems outside the scope of this change.

Reply:
That makes sense. Fixed.